### PR TITLE
Update eslint-config-universal-search version

### LINKED ISSUE #111: Adopt and enforce an ES6-based style guide
We are currently using a slightly modified version of the airbnb style guide in the idea-town JS code.

Seems to make sense to reuse the same rules here.

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #111

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,7 +1,5 @@
-extends: universal-search
+extends: universal-search/addon
 
 rules:
-  consistent-return: 0
   no-console: 0
-  no-var: 2
-  prefer-const: 2
+  func-names: 0
```
Comment (pdehaan) [inline .eslintrc:5]: [`func-names`](http://eslint.org/docs/rules/func-names.html) is a funny rule (via [**eslint-config-airbnb**](https://github.com/airbnb/javascript/blob/60480c05ea6c2d0f172302f845c39f5f62accb8d/packages/eslint-config-airbnb/rules/style.js#L19-L20)) which throws ~60 warnings in this repo if you have anonymous functions.

Oddly, I only see the warnings if I explicitly run ESLint and they don't seem to appear when I run `npm run lint` (which uses Gulp).

**Comment by jaredhirsch** [comment]
@pdehaan OK, let's do this. Mind rebasing against today's master?

**Comment by pdehaan** [comment]
@6a68 Wow, this looks harder than it should be.
I rebased and bumped and now it looks like I'm getting ~108 ESLint errors (some of which are simple style things, other errors look like it may require me to refactor to some ES6-esque class syntax).

Here's a rough sample (looks like ESLint is choking on some generator functions too):

```
➜  universal-search-addon git:(issue-111) ✗ ./node_modules/.bin/eslint .

...

/Users/pdehaan/dev/github/universal-search-addon/src/lib/main.js
    9:8   error  A space is required after '{'                                 object-curly-spacing
    9:46  error  A space is required before '}'                                object-curly-spacing
   22:1   error  Expected exception block, space or tab after "//" in comment  spaced-comment
   23:3   error  Expected exception block, space or tab after "//" in comment  spaced-comment
   27:29  error  Missing space before function parentheses                     space-before-function-paren
   28:27  error  Missing space before function parentheses                     space-before-function-paren
   29:28  error  Missing space before function parentheses                     space-before-function-paren
  114:51  error  A space is required after ','                                 comma-spacing
  123:3   error  Expected method shorthand                                     object-shorthand
  131:32  error  Missing space before function parentheses                     space-before-function-paren
  147:39  error  A space is required after '{'                                 object-curly-spacing
  147:68  error  A space is required before '}'                                object-curly-spacing
  149:5   error  Assignment to function parameter 'win'                        no-param-reassign
  207:34  error  Missing space before function parentheses                     space-before-function-paren
  246:10  error  Assignment to function parameter 'win'                        no-param-reassign
  290:16  error  Expected property shorthand                                   object-shorthand
  290:28  error  Expected property shorthand                                   object-shorthand

/Users/pdehaan/dev/github/universal-search-addon/src/lib/PlacesSearch.js
  209:31  error  Parsing error: Unexpected token *
```

I'll dig into this more tomorrow morning, or kill this PR and start from scratch and see if I can figure out what changed (looks suspiciously towards airbnb style guide).

**Comment by jaredhirsch** [comment]
@pdehaan sounds good! ping me if you have questions

**Comment by pdehaan** [comment]
Killing this in favor of my second attempt: #172

**Pull request closed without merging**